grant-project/grant-monorepo
# Milestone Enhancements

Closes #64.

### What this does
- Pulls Milestone voting actions into Proposal/Milestones tab
- Changes Milestone display to antd Steps
- Changes Proposal "Governance" tab name to "Refund" and hides it for non-contributors

### Notes
- I welcome thoughts on the layout and style, I can't see the forest for the trees at this point.
- It seems like it might be good to abstract some of the derived data into redux. `activeMilestoneIdx`, `isTrustee`, `isContributor`, `activeAccount` &etc.
- ~~I'm delaying dealing with the layout concerns of many milestones until I get some feedback on this design.~~ 

### Testing
- `yarn storybook` then look at the states under Proposal > Milestones, I think I've got most of the states there
- Visit the Proposal/Milestones tab as a contributor & trustee - go through at least one round of voting for a milestone.
- Check out responsive styles.

### Updates
After some great feedback, I've tightened things up.

- title and status line now included in left side block, really cleans up some of the visual chaos
![screen shot 2018-09-29 at 2 30 53 pm](https://user-images.githubusercontent.com/11430954/46249765-4f9e2180-c3f4-11e8-96b5-0903518462ce.png)

- selected state is filled in now (the antd "processing" style), I think it clarifies which step is selected significantly
![selected-states](https://user-images.githubusercontent.com/11430954/46249806-dfdc6680-c3f4-11e8-83e4-03234f2d3589.gif)

- step title widths are now monitored to make sure they fit, if not the active title is shown below, ant the step titles are hidden
![responsive-titles](https://user-images.githubusercontent.com/11430954/46249824-3f3a7680-c3f5-11e8-8191-6b36dd742917.gif)

- added 'geometry' stories which has from 1 to 10 milestone proposals
<img width="881" alt="screen shot 2018-09-29 at 2 39 11 pm" src="https://user-images.githubusercontent.com/11430954/46249849-9b9d9600-c3f5-11e8-8065-b4099dccd69c.png">
